src/Storages/MergeTree/MergeTreeIndexAnnoy.cpp

```diff
@@ -0,0 +1,290 @@
+#ifdef ENABLE_ANNOY
+
+#include <Storages/MergeTree/MergeTreeIndexAnnoy.h>
+
+#include <Common/typeid_cast.h>
+#include <Core/Field.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+
+namespace DB
+{
+
+namespace Annoy
```

Comment:
`namespace AnnoyIndex`? Maybe even use the same namespace as the one which contains class `ANNCondition`?